### DISCUSSION TITLE
Proposal for JSON Output Enhancements to Support Node.js API Development

### DISCUSSION THREAD

**Discussion by GiladShoham** [oxc-project/oxc#14190] [Feature Request]
Hi Team,

I'm currently developing a full Node.js API for Oxlint, which will serve as a wrapper for the CLI, complete with types and additional functionality. This will be released as an open-source project for the community once I have an initial working and documented version.

To streamline this process, it would be incredibly helpful to have more CLI commands output their results in JSON format. For example:

oxlint -V --json: This would output the version information in JSON.
oxlint --rules --json: This command could return all available rules organized by category in JSON. This feature would be particularly useful for generating TypeScript types directly from the JSON output, eliminating the need for manual copy/paste.
Additionally, it would be great to have a command like oxlint --plugins --json, which would list all available plugin names in JSON format, allowing me to generate types for plugin flags as well.

Thank you for creating such a fantastic tool. These enhancements would significantly aid in the development of the Node.js API.

Best regards,
Gilad

**Comment by DonIsaac** [comment]
Hi @GiladShoham,

A Node.js API sounds like something we should have in this repository so that it can bind directly with our Rust code without needing to parse JSON obtained via IPC. Do you have any opinions on that?

cc: @Boshen

**Comment by GiladShoham** [comment]
@DonIsaac I totally agree it's better to have it via direct binding to the rust code.
However talked to @Boshen over discord and he told me it's not planned at the moment.
Since I want to try it on real use cases I've decided to build it my own with a child process.
Once you decided to move it here I would be happy to help move my code here if you want it.
